salt/modules/aptpkg.py

```diff
@@ -495,8 +495,12 @@ def latest_version(*names, **kwargs):
 
     candidates = {}
     for line in salt.utils.itertools.split(out["stdout"], "\n"):
-        if line.endswith(":") and line[:-1] in short_names:
-            this_pkg = names[short_names.index(line[:-1])]
+        if line.endswith(":") and not re.match(r"^\s", line):
+            if line[:-1] in names:  # native package
+                this_pkg = line[:-1]
+                continue  # stop searching and evaluate details
+            if line[:-1].split(":")[0] in short_names:  # foreign-arch package by name
+                this_pkg = names[short_names.index(line[:-1].split(":")[0])]
```

Comment:
Whenever a package with multiple architectures comes up, this has the risk of overshadowing the native architecture package with the foreign one, if the latter appears later in the listing. Perhaps this should be suffixed with the package's foreign-arch (effectively becoming `line[:-1]` in the process, without changing the condition), so to avoid confusion?

Reply:
This sounds okay to me.

Reply:
To express the difference: https://github.com/saltstack/salt/compare/06ddd685fdb27e6181856bf89df1f09c1ab024db..5f73f7ee5361d31c50b16317aca7b45178b088b9

Reply:
Turns out there's a bunch of integration behavior depending on single-arg/package `latest_version()` calls returning a plain version string, so surprisingly returning a `package:version` map could cause extra breaks and scope creep beyond this bugfix, thus I've kept extra-arch matches limited to the `candidates` dict.